gadgets/trace_dns/build.yaml

```diff
@@ -0,0 +1 @@
+wasm: program.go
```

Comment:
I wanted to avoid executing Wasm in this PR because I wanted to do the refactor from #2131 first and implement the Wasm execution after the refactoring.

What do you prefer:
- I could remove this commit from this first PR and keep it for the next PR
- Make a single PR with both the "image build" Wasm feature and the Wasm execution feature